scripts/Impurity-example.jl

```diff
@@ -0,0 +1,198 @@
+
+# Now let's first do some MPI sanity checks and print some information:
+mpi_barrier() # optional, use for debugging and sanity checks
+@info "After barrier 1" mpi_rank() mpi_size() Threads.nthreads()
+
+# Now we specify parameters for constructing a two-component Hamiltonian
+P = 3 # total momentum
+m = 8 # number of modes
+na= 4 # number of non-impurity bosons
+γ = 0.2 # boson-boson coupling strength, dimensionless quantity
+η = 0.5 # impurity-boson coupling strength, dimensionless quantity
+T = m^2/2 # normalised hopping strength
+U = m*γ*na/(γ*na/(m*π^2) + 1) # converting γ to U
+V = m*η*na/(η*na/(m*π^2) + 1) # converting η to V
+# Here we use an initial address `aIni` for constructing the Hamiltonian, but 
+# it will not be used in the starting vector.
+aIni = BoseFS2C(BoseFS([na; zeros(Int, m-1)]), BoseFS([1; zeros(Int, m-1)]))
+ham = BoseHubbardMom1D2C(aIni;ta=T,tb=T,ua=U,v=V,dispersion=continuum_dispersion)
+
+
+
+# Now we can setup the Monte Carlo parameters
+steps_warmup = 10_000 # number of QMC steps running with a dummy Hamiltonian, see Stage 1
+steps_equilibrate = 10_000 # number of QMC steps running with the real `ham`
+steps_final = 10_000 # number of QMC steps running with G2 correlators, very slow, be caution!
+tw = 1_000 # number of walkers, be sure to use a larger enough number to eliminate biases
+
+# Specifying the shift strategy:
+s_strat = DoubleLogUpdateAfterTargetWalkers(targetwalkers = tw)
+
+# Wrapping `dv` for MPI:
+dv = MPIData(init_dv(P,m,na));
+
+# Let's have a look of the starting vector, in this particular case, all 4 different ways of 
+# distributing total momenta `P` with `init_dv()` are triggered:
+@mpi_root @show dv
+
+# ## Stage 1: Running with the "dummy" Hamiltonian
+
+# Here we are constructing a secondary Hamiltonian `ham2` with equal boson-boson and impurity coupling
+# strength. We use this Hamiltonian to further generate a batter staring vector. From previous experiences 
+# calculating impurity problems, this setup can significantly speed up the convergence and help FCIQMC to 
+# sample the important part of the Hilbert space, especially useful when `η` is very small.
+η2 = γ
+V2 = m*η2*na/(η2*na/(m*π^2) + 1)
+ham2 = BoseHubbardMom1D2C(aIni;ta=T,tb=T,ua=U,v=V2,dispersion=continuum_dispersion)
+
+# We use very small time-step size and high starting value of shift
+params = RunTillLastStep(step = 0, dτ = 0.00001, laststep = steps_warmup,shift = 200.0)
+# As we only use the secondary Hamiltonian `ham2` to generate a staring vector, we don't have to
+# save any data in this stage. Progress messages are suppressed with `io=devnull`.
+r_strat = ReportDFAndInfo(reporting_interval = 1_000, info_interval = 1_000, writeinfo = is_mpi_root(), io = devnull)
+
+# Now we run FCIQMC with `lomc!()` and track the elapsed time. 
+# Both `df` and `state` will be overwritten later with the "real" data.
+el = @elapsed df, state = lomc!(ham2, dv; params, s_strat, r_strat,)
```

Comment:
```suggestion
el = @elapsed lomc!(ham2, dv; params, s_strat, r_strat,)
```
If we are not interested in `df` and `state`, why ask to have them returned in the first place?

Reply:
Good point, will remove them.